demo/api/src/auth/auth.module.ts

```diff
@@ -21,28 +21,45 @@ export const SYSTEM_USER_NAME = "system-user";
 @Module({})
 export class AuthModule {
     static forRoot(config: Config): DynamicModule {
+        const authServices = [
+            createBasicAuthService({
+                username: SYSTEM_USER_NAME,
+                password: config.auth.systemUserPassword,
+            }),
+            createSitePreviewAuthService({ sitePreviewSecret: config.sitePreviewSecret }),
+        ];
+
+        const providers: Provider[] = [
+            createAuthResolver({
+                postLogoutRedirectUri: config.auth.postLogoutRedirectUri,
+                endSessionEndpoint: config.auth.idpEndSessionEndpoint,
+            }),
+            UserService,
+            AccessControlService,
+            {
+                provide: APP_GUARD,
+                useClass: CometAuthGuard,
+            },
+        ];
+
+        if (config.auth.useAuthProxy) {
+            authServices.push(
+                createJwtAuthService({ verifyOptions: { secret: "secret" } }), // for testing purposes, send header "Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIyIiwiaWF0IjoxNTE2MjM5MDIyfQ.fG9j2rVOgunoya_njgn9w1t8muFlrpE9ffJ9i8sJYsQ"
+            );
```

Comment:
@fraxachun is this something we should add to demo? Shouldn't we set jwksUri etc. instead?